addon/components/date-picker.js

```diff
@@ -1,44 +1,30 @@
 import LazyTextField from '../components/lazy-text-field';
-
-const A_DAY = 1000 * 60 * 60 * 24;
-
-function today() {
-  let date = new Date();
-  date.setHours(0);
-  date.setMinutes(0);
-  date.setSeconds(0);
-  date.setMilliseconds(0);
-
-  return date;
-}
+import moment        from 'moment';
+import { get, set }  from '@ember/object';
 
 export default LazyTextField.extend({
   classNames: ['date-picker'],
+  dateFormat: 'DD-MM-YYYY',
 
   didReceiveAttrs() {
-    this.set('date', this.getAttr('value'));
+    set(this, 'date', this.getAttr('value'));
     this._super(...arguments);
   },
 
   keyDown(e) {
-    let value = this.get('date');
+    let value = moment(get(this, 'date'));
+    let scale = e.shiftKey ? 'month' : 'day';
 
     if ([38, 40].indexOf(e.keyCode) !== -1) {
       this.withLazyDisabled(() => {
-        if (e.keyCode === 38) {
-          if (e.shiftKey) {
-            this.send('changed', new Date(value.setMonth(value.getMonth() + 1)));
-          } else {
-            this.send('changed', new Date(+value + A_DAY));
-          }
+        if (e.keyCode === 38) { // arrow up
+          value = value.add(1, scale);
+          return this.send('changed', value);
```

Comment:
There are various ways to make these if statements shorter
1. return value directly
2. put diff or the function in a variable, and call `changed` once afterwards